### ISSUE TITLE
Add Softmax and Cross-Entropy

### ISSUE THREAD

**Issue by adriannosampaio** [adriannosampaio/SimpleNeuralNetwork#23]
- Add Softmax activation function
- Add Cross-Entropy loss
- Extract Loss functions to new classes/file
